python/lsst/ip/isr/isrFunctions.py

```diff
@@ -546,10 +560,30 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
         stdevBiasArr = 0.74*(percentiles[2] - percentiles[0])  # robust stdev
         diff = numpy.abs(biasArray - medianBiasArr[:, numpy.newaxis])
         biasMaskedArr = numpy.ma.masked_where(diff > numSigmaClip*stdevBiasArr[:, numpy.newaxis], biasArray)
-        collapsed = numpy.mean(biasMaskedArr, axis=1)
-        if collapsed.mask.sum() > 0:
-            collapsed.data[collapsed.mask] = numpy.mean(biasArray.data[collapsed.mask], axis=1)
+
+        if fitType in ('MEDIAN_PER_ROW',):
+            collapsed = []
+            fitTypeStats = afwMath.stringToStatisticsProperty('MEDIAN')
+            # Loop over rows because there is no afwMath.Median(array, axis=1)
+            for row in biasMaskedArr:
+                if row.mask is False:
+                    row.mask = numpy.repeat(False, len(row))
+                medianRow = afwMath.makeStatistics(row, row.mask, fitTypeStats, statControl).getValue()
```

Comment:
Also, I just realised, doing this here means you're actually doing a single-round-sigma-clipped row-wise median, right? Not saying that's bad _per se_ but trying to understand what's happening here.

Reply:
There are a few lines before the loop that do some clipping, and then, in the loop, the median is taken, as you point out. This section of the code was originally intended for the polynomial cases of the overscan correction, so it might be redundant in the case of the median per row.  But then if I move the loop to the other section, things might break as we pointed out in another comment.

Reply:
Can you check with @RobertLuptonTheGood (or he can confirm here, now that he's tagged) that he's OK with doing a single round of clipping before doing the per-row median please? I don't it's necessarily bad, but maybe there is some statistical subtlety that will cause him to dislike this.